.github/workflows/integration-test.yml

```diff
@@ -0,0 +1,70 @@
+# This workflow will build a Java project with Gradle
+# For more information see: https://help.github.com/actions/language-and-framework-guides/building-and-testing-java-with-gradle
+
+name: Run Integration Tests
+
+on:
+  push:
+    branches:
+      - master
```

Comment:
I think it's also worth running the tests on non-master branch commits. We'd want to detect a code issue (if there's one) **before** it gets to master, IMO.

Reply:
I agree, there will be a big effort to refactor this workflow once we have on-demand MC-Terra/FiaB integration working. Then we can run this on every PR and prevent merge if the tests fail.

I think the original motivation for just running against master is to not have multiple PRs running tests against the same environment at the same time. Also, are the tests ever going to conflict with each other if run concurrently from multiple PRs? (I haven't audited the existing test but that's something we'd need to keep in mind for future ones until we have on-demand MC-Terras)

Reply:
The existing test does create a workspace in a self-contained manner that shouldn't conflict with say, another CreateWorkspace request, in theory. This has not been test out.

We currently know of the CreateWorkspace quota limitation (about 70 requests/10 seconds) that will be tackled in AS-299, but it's probably unlikely that we'll have enough concurrent PR's to run into that issue for integration tests.
That said, I don't know whether there are other limitations for concurrent runs. But in any case - we should always aim to build the tests out in a way that allows concurrent runs.

But yes, we can start with a workflow for only master and refactor later to include PRs.